core/types/src/target_info.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+//! SGX TargetInfo
+
+use crate::{
+    config_id::ConfigId, new_type_accessors_impls, Attributes, ConfigSvn, Measurement,
+    MiscellaneousSelect, MrEnclave,
+};
+use mc_sgx_core_sys_types::{
+    sgx_target_info_t, SGX_TARGET_INFO_RESERVED1_BYTES, SGX_TARGET_INFO_RESERVED2_BYTES,
+    SGX_TARGET_INFO_RESERVED3_BYTES,
+};
+
+/// The target info
+#[repr(transparent)]
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct TargetInfo(sgx_target_info_t);
+
+impl TargetInfo {
+    /// The MRENCLAVE measurement
+    pub fn mr_enclave(&self) -> Measurement {
+        Measurement::MrEnclave(self.0.mr_enclave.into())
+    }
+
+    /// The attributes
+    pub fn attributes(&self) -> Attributes {
+        self.0.attributes.into()
+    }
+
+    /// The Config SVN
+    pub fn config_svn(&self) -> ConfigSvn {
+        self.0.config_svn.into()
+    }
+
+    /// Miscellaneous Select values
+    pub fn miscellaneous_select(&self) -> MiscellaneousSelect {
+        self.0.misc_select.into()
+    }
+
+    /// The Config ID
+    pub fn config_id(&self) -> ConfigId {
+        self.0.config_id.into()
+    }
+}
+
+new_type_accessors_impls! {
+    TargetInfo, sgx_target_info_t;
+}
+
+impl Default for TargetInfo {
+    fn default() -> Self {
+        Self(sgx_target_info_t {
+            mr_enclave: MrEnclave::default().into(),
+            attributes: Attributes::default().into(),
+            reserved1: [0u8; SGX_TARGET_INFO_RESERVED1_BYTES],
+            config_svn: ConfigSvn::default().into(),
+            misc_select: MiscellaneousSelect::default().into(),
+            reserved2: [0u8; SGX_TARGET_INFO_RESERVED2_BYTES],
+            config_id: ConfigId::default().into(),
+            reserved3: [0u8; SGX_TARGET_INFO_RESERVED3_BYTES],
+        })
+    }
+}
```

Comment:
This should probably get deleted eventually (though `sgx_target_info_t` should keep it) in favor of a monkey-patched API in `mc-sgx-urts`:

```rust
pub trait TryFromId {
    fn try_from_id(id: EnclaveId) -> Self;
}

impl TryFromId for TargetInfo {
    fn try_from_id( id: EnclaveId) -> Result<Self, SgxError> {
        let mut info = sgx_target_info_t::default();
        unsafe{ sgx_get_target_info(id, &mut info as *mut _) }
            .into_result()?
            .map(|_| TargetInfo::from(info))
}
```

Reply:
the Default for `TargetInfo` goes away in #115